src/lib/iproute2_cmdgen.c

```diff
@@ -10,12 +10,251 @@
  *
  * Copyright (C) 2024 Okda Networks, <aaqrbaw@okdanetworks.com>
  */
+#include <ctype.h>
 
 #include "iproute2_cmdgen.h"
 
+#define CMD_LINE_SIZE 1024
+
+typedef enum {
+    ADD,
+    DELETE,
+    UPDATE,
+    UNKNOWN,
+} oper_t;
+
+
+void copy_argv(char ***dest, char **src, int argc)
+{
+    *dest = (char **) malloc(argc * sizeof(char *));
+    if (*dest == NULL) {
+        fprintf(stderr, "Memory allocation failed\n");
+        exit(EXIT_FAILURE);
+    }
+    for (int i = 0; i < argc; i++) {
+        (*dest)[i] = strdup(src[i]);
+        if ((*dest)[i] == NULL) {
+            fprintf(stderr, "Memory allocation failed\n");
+            exit(EXIT_FAILURE);
+        }
+    }
+}
+
+
+void free_argv(char **argv, int argc)
+{
+    // Free memory for each string
+    for (int i = 0; i < argc; i++) {
+        free(argv[i]);
```

Comment:
even if argv will be free'd, it'd be nice to set argv[i] = null just in case someone keeps a reference.